packages/amplify-graphql-schema-generator/src/__tests__/schema-overrides.test.ts

```diff
@@ -397,6 +398,229 @@ describe('apply schema overrides for models with refersTo', () => {
   });
+
+  it('should not allow duplicate model name mappings in edited schema', () => {
+    const generatedSchemas = [
+      `
+        type Post @model {
+            id: String! @primaryKey
+            content: String
+        }
+      `,
+      `
+        type MyPost @model @refersTo(name: "Post") {
+            id: String! @primaryKey
+            content: String
+        }
+      `,
+    ];
+    const editedSchema = `
+        type Post @model {
+            id: String! @primaryKey
+            content: String
+        }
+        type MyPost @refersTo(name: "Post") @model {
+            id: String! @primaryKey
+            content: String
+        }
+    `;
+    const editedDocument = parse(editedSchema);
+    generatedSchemas.forEach((generatedSchema) => {
+      const document = parse(generatedSchema);
+      expect(() => applySchemaOverrides(document, editedDocument)).toThrowErrorMatchingInlineSnapshot(
+        `"Models Post, MyPost are mapped to the same table Post. Remove the duplicate mapping."`,
+      );
+    });
+  });
+});
+
+describe('apply schema overrides for model fields with refersTo', () => {
+  it('should be the same if no edits are made', () => {
+    const schema = `
+        type Post @refersTo(name: "posts") @model {
+            id: ID!
+            nameField: String @refersTo(name: "name_field")
+        }
+        `;
+    const document = parse(schema);
+    const updatedDocument = applySchemaOverrides(document, document);
+    stringsMatchWithoutWhitespace(print(updatedDocument), schema);
+  });
+
+  it('should allow field name overrides', () => {
+    const document = parse(`
+              type Post @refersTo(name: "posts") @model {
+                  id: ID!
+                  nameField: String @refersTo(name: "name_field")
+              }
+          `);
+    const editedSchema = `
+              type MyPost @refersTo(name: "posts") @model {
+                  id: ID!
+                  name: String @refersTo(name: "name_field")
+              }
+          `;
+    const editedDocument = parse(editedSchema);
+    const updatedDocument = applySchemaOverrides(document, editedDocument);
+    stringsMatchWithoutWhitespace(print(updatedDocument), editedSchema);
+  });
+
+  it('should allow removing inferred field name mappings', () => {
+    const document = parse(`
+              type Post @model {
+                  id: ID! @refersTo(name: "my_id")
+                  nameField: String @refersTo(name: "name_field")
+              }
+          `);
+    const editedSchema = `
+              type Post @model {
+                  id: ID! @refersTo(name: "my_id")
+                  name_field: String
+              }
+          `;
+    const editedDocument = parse(editedSchema);
+    const updatedDocument = applySchemaOverrides(document, editedDocument);
+    stringsMatchWithoutWhitespace(print(updatedDocument), editedSchema);
+  });
+
+  it('should allow adding refersTo to fields', () => {
```

Comment:
Can you add a test for field level conflicts (2 GraphQL fields referring to the same SQL field)